src/SNMP_Agent.cpp

```diff
@@ -86,72 +84,110 @@ SortableOIDType* SNMPAgent::buildOIDWithPrefix(char* oid, bool overwritePrefix){
     return nullptr;
 }
 
-ValueCallback* SNMPAgent::addReadWriteStringHandler(char* oid, char** value, size_t max_len, bool isSettable, bool overwritePrefix){
+ValueCallback* SNMPAgent::addReadWriteStringHandler(const char *oid, char** value, size_t max_len, bool isSettable, bool overwritePrefix){
     if(!value || !*value) return nullptr;
 
     SortableOIDType* oidType = buildOIDWithPrefix(oid, overwritePrefix);
     if(!oidType) return nullptr;
     return addHandler(new StringCallback(oidType, value, max_len), isSettable);
 }
 
-ValueCallback *SNMPAgent::addReadOnlyStaticStringHandler(char *oid, std::string value, bool overwritePrefix) {
+ValueCallback *SNMPAgent::addReadOnlyStaticStringHandler(const char *oid, const std::string& value, bool overwritePrefix) {
     SortableOIDType* oidType = buildOIDWithPrefix(oid, overwritePrefix);
     if(!oidType) return nullptr;
     return addHandler(new ReadOnlyStringCallback(oidType, value), false);
 }
 
-ValueCallback* SNMPAgent::addOpaqueHandler(char* oid, uint8_t* value, size_t data_len, bool isSettable, bool overwritePrefix){
+
+ValueCallback* SNMPAgent::addOpaqueHandler(const char *oid, uint8_t* value, size_t data_len, bool isSettable, bool overwritePrefix){
```

Comment:
I believe making the oid a const char* will remove the ability to dynamically create OID's when setting up the agent.
I have some projects where I need to dynamically create OID handlers on the fly, that are not known at compile time, so I think this will need to stay as `char *`.
Thoughts?

Reply:
The compiler automatically converts `char*` to `const char*`. This is absolutely safe and acceptable (you won't get compilation error). You can pass both const strings and non-const char tables to function with `const char*` argument.

Opposite (`const char*` to `char*`) is treated as an error by any modern compiler (unless you cast to `char*`).
With `const char*` oid type both usages are valid:
```C
// non-const dynamic string
char tmp[32];
sprintf(tmp, "Hello");
addOpaqueHandler(tmp,...); //OK

// const string
addOpaqueHandler("Hello",...); //OK
```
The only scenario you would like `char*` oid type is when the function modifies oid. As far as I know this is not the case (correct me if I'm wrong).

Reply:
Ah of course, You've quite right. Sorry I clearly had a brain fart.
Thanks for the detailed explanation!